### ISSUE TITLE
GH connection issue - connection required after space creation

### ISSUE THREAD

**Issue by mishaone** [openshiftio/openshift.io#2718]
I'm testing the GitHub connect part by disconnecting my GitHub account and seeing what it lets me do without prompting for the GitHub connection. IMO, this should probably happen when we create the account as before because it's causing some weird issues to pop up. For example:

The one time it let me create a space without GH, it stopped me at the quickstart creation point to ask for GH credentials. Not sure which point is by design and which is an error. Should it be asking for GH connection at space creation or after?

**Comment by alexeykazakov** [comment]
It should ask you to connect GitHub, so it can create a space.
@catrobson @dgutride is there any design/flow of this? We should not silently initiate GH linking. We should ask user to connect GH account so it's clear what we are doing.

cc: @rohitkrai03

**Comment by dgutride** [comment]
@alexeykazakov  - the space wizard won't launch if you aren't connected to github.  It depends on how the user disconnects, though - just deauthorizing the account doesn't tell openshift.io anything until you try to use the connect.  Disconnecting using the disconnect button in the UI will tell the user they need to connect to github.  @mishaone - this behavior is by design

**Comment by mishaone** [comment]
@dgutride - is there information around what the disconnect from GitHub button is meant to do, exactly? I don't know what the user story is, so assume it's something like what I tried to do yesterday (disconnect from GH account #1 and connect to a different one) but it sounds like you can't do that by design.

**Comment by alexeykazakov** [comment]
@mishaone as said before you have to log-out from GitHub first if you want to connect to another GH account.
We can't control GitHub login. When you connect your account we just redirect you to GH login. But if you already logged into GH then you won't see the login page.

And I just tried it myself I don't see any issue with the wizard.
If I disconnect GH and then try to launch the wizard I see:

![screenshot from 2018-03-21 18-59-37](https://user-images.githubusercontent.com/620087/37747091-a0747088-2d3a-11e8-8dda-88bdd4065433.png)

I thought that the issue you are reporting is that we don't show this dialog ^ and just connect GH automatically. But it's not the case.

So, what's the problem? Can you make a screencast or something?

**Comment by mishaone** [comment]
The issue is a bit interlaced because of splitting it up, but in this case, the issue was that it let me create a space without GitHub once out of three times I tested and then asked me for GitHub when I tried to create a quickstart using the wizard. If the correct place to ask for GH is when you create a space, then it should not let us create a space without GH. I can try to reproduce this but it was mostly just a combination of disconnecting and then trying to create a space.
